### PR TITLE
Protocol v1.0

### PR DESCRIPTION
**Pull request by fmoor**

### CODE REVIEW

File: granularflow1pX.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+}
+
+func (c *protocolConnection) optimistic1pX(
+	r *buff.Reader,
+	q *query,
+	cdcs *codecPair,
+) error {
+	// When descriptors are returned the codec ids sent didn't match the
+	// server's.  The codecs should be rebuilt with the new descriptors and the
+	// execution retried.
```
Comment (elprans) [inline granularflow1pX.go:72]: This is not correct.  When the output type descriptor is not matched, the client is supposed to receive the correct one and proceed decoding the data messages, no execution retry is necessary.

Comment (fmoor) [reply granularflow1pX.go:72]: Ok. What about if the input descriptor is wrong? Does that warrant an execution retry?

Comment (elprans) [reply granularflow1pX.go]: For now, no, because it's a much rarer condition and is likely non-recoverable anyway.

**Review by fantix — COMMENTED** [review]
The API to set aliases/globals/config in the state is a TODO, right?

**Comment by fmoor** [comment]
> The API to set aliases/globals/config in the state is a TODO, right?

added

**Review by elprans — APPROVED** [review]

**Pull request merged**